### PR TITLE
bpo-32392: Document env kwarg of subprocess.run method

### PR DESCRIPTION
**Pull request by rixx**
As discussed in [bpo-32392](https://bugs.python.org/issue32392), I added the `env` kwarg to the subprocess.run documentation.

https://bugs.python.org/issue32392

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


When your account is ready, please add a comment in this pull request
and a Python core developer will remove the `CLA not signed` label
to make the bot check again.

Thanks again to your contribution and we look forward to looking at it!

File: Doc/library/subprocess.rst
Comment (berkerpeksag) [inline Doc/library/subprocess.rst:42]: Could you add a brief description of what `env` is? Copying the existing documentation would be enough:

```rst
   If *env* is not ``None``, it must be a mapping that defines the environment
   variables for the new process; these are used instead of the default
   behavior of inheriting the current process' environment.
```

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by rixx** [comment]
Thank you for your reveiw. I have made the requested changes; please review again. I've added a sentence referencing `Popen` so that people will be more likely to find the note attached to the `env` kwarg of `Popen`, copying it would be overkill I think.

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Looks good to me, thank you! I will wait until Monday to give @gpshead time to review this since he's the maintainer of the subprocess module.

**Comment by berkerpeksag** [comment]
Thanks!

**Comment by berkerpeksag** [comment]
@rixx would you like to submit a backport PR for 3.6?

**Pull request merged**
